npm/translatePolicy.go

```diff
@@ -423,21 +473,129 @@ func translateIngress(ns string, policyName string, targetSelector metav1.LabelS
 			}
 
 			if fromRule.PodSelector == nil && fromRule.NamespaceSelector != nil {
-				nsLabelsWithOps, _, _ := parseSelector(fromRule.NamespaceSelector)
-				_, nsLabelsWithoutOps := GetOperatorsAndLabels(nsLabelsWithOps)
-				if len(nsLabelsWithoutOps) == 1 && nsLabelsWithoutOps[0] == "" {
-					// Empty namespaceSelector. This selects all namespaces
-					nsLabelsWithoutOps[0] = util.KubeAllNamespacesFlag
-				} else {
-					for i, _ := range nsLabelsWithoutOps {
-						// Add namespaces prefix to distinguish namespace ipset lists and pod ipsets
-						nsLabelsWithoutOps[i] = "ns-" + nsLabelsWithoutOps[i]
+				for _, nsSelector := range FlattenNameSpaceSelector(fromRule.NamespaceSelector) {
```

Comment:
Prefer to splitting this one line to two lines to understand what it returns. Same to all in this file.